privcount/protocol.py

```diff
@@ -399,6 +509,46 @@ def lineReceived(self, line): # overrides twisted function
                         self.factory.stop_injecting()
                 else:
                     self.sendLine('552 Unrecognized event ""')
+            elif parts[0] == "GETINFO":
+                # the correct response to an empty GETINFO is an empty OK
+                if len(parts) == 1:
+                    self.sendLine("250 OK")
+                # GETINFO is case-sensitive, and returns multiple lines
+                # The control spec says we should say "Tor ", but tor doesn't
+                elif len(parts) == 2 and parts[1] == "version":
+                    self.sendLine("250-version=0.2.8.6 (git-4d217548e3f05569)")
+                    self.sendLine("250 OK")
+                elif len(parts) == 2 and parts[1] == "address":
+                    # TEST-NET from https://tools.ietf.org/html/rfc5737
+                    self.sendLine("250-address=192.0.2.91")
+                    self.sendLine("250 OK")
+                elif len(parts) == 2 and parts[1] == "fingerprint":
+                    self.sendLine("250-fingerprint=FACADE0000000000000000000123456789ABCDEF")
+                    self.sendLine("250 OK")
+                else:
+                    # strictly, GETINFO should process each word on the line
+                    # as a separate request for information.
+                    # But this is sufficient for our purposes, and is still
+                    # more-or-less conformant to the control spec.
+                    self.sendLine('552 Unrecognized key "{}"'.format(parts[1]))
+            elif parts[0] == "GETCONF":
+                # the correct response to an empty GETCONF is an empty OK
+                if len(parts) == 1:
+                    self.sendLine("250 OK")
+                # Unlike GETINFO, GETCONF is case-insensitive, and returns one line
+                # It also uses the canonical case of the option in its response
+                elif len(parts) == 2 and parts[1].lower() == "nickname":
+                    self.sendLine("250 Nickname=PrivCountTorRelay99")
+                elif len(parts) == 2 and parts[1].lower() == "orport":
+                    self.sendLine("250 ORPort=9001")
+                elif len(parts) == 2 and parts[1].lower() == "dirport":
+                    self.sendLine("250 DirPort=9030")
+                else:
+                    # strictly, GETCONF should process each word on the line
```

Comment:
Oops, that's what copy and paste gets you.
I summarised the second comment in both branches containing that commit:
[results-add-context 12d9030] fixup! Add nickname, orport, dirport, version, and address to context
[add-dc-last-event 23bba08] fixup! Add nickname, orport, dirport, version, and address to context